### PR TITLE
Fix button positioning for load-more elements as not to partially cover last item in amp-list

### PR DESCRIPTION
**Pull request by cathyxz**
Part of https://github.com/ampproject/amphtml/issues/14060. 
Demo: https://amp-infinite-scroll.glitch.me/
~Still need to fix a small bug regarding `classList.toggle` not working in the last `setLoadMore` call to hide the see more button.~
Fixed.

### CODE REVIEW

File: extensions/amp-list/0.1/amp-list.js [hunk 1]
```diff
@@ -495,7 +495,8 @@ export class AmpList extends AMP.BaseElement {
       this.templates_.findAndRenderTemplateArray(this.element, array)
           .then(results => this.updateBindings_(results))
           .then(elements => this.render_(elements, current.append))
-          .then(() => this.maybeRenderLoadMoreTemplates_(payload))
+          .then(() => this.loadMoreEnabled_
```
Comment (cvializ) [inline extensions/amp-list/0.1/amp-list.js]: signal: the `.then(() => booleanValue && this.promiseReturningFunction())` works, but could be leaning in the direction of being unhelpfully terse

Comment (cathyxz) [reply extensions/amp-list/0.1/amp-list.js]: I agree. Let me refactor this then instead of reusing this odd pattern.

Comment (cvializ) [reply extensions/amp-list/0.1/amp-list.js]: sg, feel free to refactor in a separate PR if this fixes a bug for your feature

Comment (cathyxz) [reply extensions/amp-list/0.1/amp-list.js]: Done.

**Pull request merged**